lib/winapi/thread.c

```diff
@@ -89,6 +89,22 @@ VOID ExitThread (DWORD dwExitCode)
     PsTerminateSystemThread(dwExitCode);
 }
 
+BOOL GetExitCodeThread (HANDLE hThread, LPDWORD lpExitCode)
+{
+    PETHREAD thread;
+    NTSTATUS status;
+
+    status = ObReferenceObjectByHandle(hThread, NULL, (PVOID *)&thread);
+    if (!NT_SUCCESS(status)) {
+        SetLastError(RtlNtStatusToDosError(status));
+        return FALSE;
+    }
+
+    *lpExitCode = thread->ExitStatus;
```

Comment:
Is this initialized to `STILL_ACTIVE`? If not, should we do this, or do we need to tag threads differently?

Reply:
Ah, this somehow managed to slip by me. We probably shouldn't modify this kernel data, I fixed it by checking `thread->Tcb.HasTerminated` and returning `STILL_ACTIVE` when it's not true. I checked the result of `GetExitCodeThread` (both return value and exit code) before and after the thread terminated, and it works fine now. I also compared it to Windows 7 now to make sure it's supposed to return true even when the thread hasn't terminated yet.